packages/time/src/utils/getFirstDayOfWeek.ts

```diff
@@ -0,0 +1,8 @@
+import { Temporal } from '@js-temporal/polyfill'
+
+export const getFirstDayOfWeek = (currWeek: string, locale: Intl.UnicodeBCP47LocaleIdentifier | Intl.Locale = 'en-US') => {
+  const date = Temporal.PlainDate.from(currWeek);
+  const loc = new Intl.Locale(locale);
+  const { firstDay } = loc.weekInfo || loc.getWeekInfo();
```

Comment:
With redesign of the polyfil you would only need to call `loc.getWeekInfo()` and it would work the same across all browsers.